### PR TITLE
Add example with parameters

### PR DESCRIPTION
**Pull request by tomanagle**
- Replaces the PI example with an example that uses parameters to show that the key must be unique for each resource
- Adds an extra line of explainer to the key option in the readme and common.ts

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -69,12 +69,36 @@ console.log(await getPi());
 // that now replaces current cache entry and is cached for 5 minutes
 ```
```
Comment (Xiphe) [inline readme.md:70]: I'd get rid of this example in favor of yours

File: readme.md [hunk 2]
```diff
@@ -69,12 +69,36 @@ console.log(await getPi());
...
+// lru cache is not part of this package but a simple non-persistent cache
+const lru = new LRUCache<string, CacheEntry<string>>({ max: 1000 });
+
+function getUserById({ userId }: { userId: string }): Promise<User> {
+  return cachified({
+    // Use the userId as part of the key to make sure we cache the right user
+    key: `users_${userId}`,
+    cache: lru,
+    async getFreshValue() {
+     const response fetch(`https://example.com/users/${userId}`)
```
Comment (Xiphe) [inline readme.md:87]: theres an `=` missing and indentation is wrong

File: readme.md [hunk 3]
```diff
@@ -69,12 +69,36 @@ console.log(await getPi());
...
+    return response.json();
```
Comment (Xiphe) [inline readme.md:88]: please add two more spaces for indentation

**Review by Xiphe — CHANGES_REQUESTED** [review]
Awesome! Thanks for the contribution and input. I have some minor changes I'd like to see with this. If you want I can also add them, just let me know.

**Comment by tomanagle** [comment]
Nah, all good. I'll update the PR now. Thanks for the feedback

**Comment by tomanagle** [comment]
I've updated the PR. I removed the PI example and added the `getUserById` example. I also changed the URL to something that will work, just because people may want to copy it and run it to see t working.

**Comment by Xiphe** [comment]
awesome! thank you for the contribution 🎉

**Pull request merged**